### ISSUE TITLE
java.lang.IllegalStateException: Source entity has no ID (should have been put before)

### ISSUE THREAD

**Issue by indrakumarprajapat** [objectbox/objectbox-java#372]
Here is my Item class,

```
@Entity
public class Item implements Serializable { //extends Model

    @Id(assignable = true)
    private long id;

    public ToMany<Tax> taxes_list;
    ...
}
```
The app crashing everytime, when we add new Tax into taxes_list, if the Item object is already in database. i mean, if I get Item object from db and then add new Tax in taxes_list and then update the Item object, then app crashes. I already tried the solution, 

```
toMany.clear();
toMany.addAll(newList);
```

**Ref:**
https://stackoverflow.com/questions/47247670/objectbox-source-entity-has-no-id-should-have-been-put-before
 
But  its not working. crashing the app.

**Exception log:**
```
FATAL EXCEPTION: main Process: com.example, PID: 16767
java.lang.IllegalStateException: Source entity has no ID (should have been put before)
    at io.objectbox.relation.ToMany.internalCheckApplyToDbRequired(ToMany.java:599)
    at com.example.entity.TaxCursor.put(TaxCursor.java:72)
    at com.example.entity.TaxCursor.put(TaxCursor.java:17)
    at io.objectbox.relation.ToMany.internalApplyToDb(ToMany.java:694)
    at com.example.entity.ItemCursor.put(ItemCursor.java:95)
    at com.example.entity.ItemCursor.put(ItemCursor.java:17)
    at io.objectbox.Box.put(Box.java:355)
    at com.example.repository.ItemRepo.saveItemForTaxes(ItemRepo.java:82)
    at com.example.Activities.tax.ITTaxCreateActivity.updateItemsEnabilityInLocalDb(ITTaxCreateActivity.java:276)
    at com.example.Activities.tax.ITTaxCreateActivity.access$700(ITTaxCreateActivity.java:40)
    at com.example.Activities.tax.ITTaxCreateActivity$7.onSuccess(ITTaxCreateActivity.java:348)
    at android.os.Handler.handleCallback(Handler.java:751)
    at android.os.Handler.dispatchMessage(Handler.java:95)
    at android.os.Looper.loop(Looper.java:154)
    at android.app.ActivityThread.main(ActivityThread.java:6165)                                                                   
    at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:888)
    at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:778)


```

**Comment by greenrobot-team** [comment]
Is this possibly related to the issue [described in the docs](http://objectbox.io/documentation/relations/)? If so try to `box.attach(item)` before modifying a ToMany.

Note: if yes, this is internal issue `open-source/ticket/38`.
-ut

**Comment by greenrobot-team** [comment]
Disregard my previous comment. This actually looks like #296. This should be fixed as of version `1.4.1`.

What version of ObjectBox are you using?
-ut

**Comment by greenrobot-team** [comment]
No response. Closing. -ut
